required.go

```diff
@@ -6,35 +6,73 @@ import (
 	"strings"
 )
 
+type Option uint8
+
+const (
+	// Allows empty string, array or map. Default is false
+	AllowEmptyString Option = iota
+	AllowEmptyArray
+	AllowEmptyMap
+	AllowEmptySlice
+)
+
 // Required Validator is very simple
-type RequiredValidator struct{}
+type Required struct {
+	opts map[Option]bool
+}
 
-// IsValid determines if the given value satisfies a "Required" rule.
-func (r *RequiredValidator) IsValid(value interface{}) (bool, []error) {
+func (r *Required) SetOpts(opts map[Option]bool) {
+	r.opts = opts
+}
 
+// IsValid determines if the given value satisfies a "Required" rule.
+func (r *Required) IsValid(value interface{}) (bool, []error) {
 	if value == nil {
-		return false, singleErrorSlice("nil is not a valid value for RequiredValidator")
+		return false, singleErrorSlice("nil is not a valid value for Required")
```

Comment:
"Base-level" here is also an example, you could also say "Object" or "General", etc